PhysicsTools/NanoAOD/python/nanogen_cff.py

```diff
@@ -0,0 +1,151 @@
+from PhysicsTools.NanoAOD.taus_cff import *
+from PhysicsTools.NanoAOD.jets_cff import *
+from PhysicsTools.NanoAOD.globals_cff import genTable
+from PhysicsTools.NanoAOD.met_cff import metMCTable
+from PhysicsTools.NanoAOD.genparticles_cff import *
+from PhysicsTools.NanoAOD.particlelevel_cff import *
+from PhysicsTools.NanoAOD.lheInfoTable_cfi import *
+from PhysicsTools.NanoAOD.genWeightsTable_cfi import *
+from PhysicsTools.NanoAOD.common_cff import CandVars
+
+nanoMetadata = cms.EDProducer("UniqueStringProducer",
+    strings = cms.PSet(
+        tag = cms.string("untagged"),
+    )
+)
+
+nanogenSequence = cms.Sequence(
+    nanoMetadata+
+    particleLevel+
+    genJetTable+
+    patJetPartons+
+    genJetFlavourAssociation+
+    genJetFlavourTable+
+    genJetAK8Table+
+    genJetAK8FlavourAssociation+
+    genJetAK8FlavourTable+
+    tauGenJets+
+    tauGenJetsSelectorAllHadrons+
+    genVisTaus+
+    genVisTauTable+
+    genTable+
+    genParticleTables+
+    tautagger+
+    rivetProducerHTXS+
+    particleLevelTables+
+    metMCTable+
+    genWeightsTable+
+    lheInfoTable
+)
+
+def nanoGenCommonCustomize(process):
+    process.rivetMetTable.extension = False
+    process.lheInfoTable.storeLHEParticles = True
+    process.lheInfoTable.precision = 14
```

Comment:
you should add the process.genWeightsTable.keepAllPSWeights = True
this was done in master in cms-sw#31276

Reply:
Good point, thanks